### ISSUE TITLE
Redirect link after delete does not accept regular typolink parameters

### ISSUE THREAD

**Issue by rowild** [in2code-de/femanager#315]
A configuration like this:

```
	settings {
		delete {
			# redirect after profile delete
			redirect = TEXT
			redirect {
				typolink {
					parameter = 1
					returnLast = url
					forceAbsoluteUrl = 1
					additionalParams = &accountDeleted=1
					#linkAccessRestrictedPages = 1
					target = _blank
					extTarget = _blank
					addQueryString = 1
				}
			}
		}
	}

```

only respects "parameter" and "returnLast". 
If this is the expected behaviour, it should be mentioned in the docu. If not, then I assume there is a bug somewhere on the run?

**Comment by greggandorf** [comment]
Also the redirect does not work even with parameter and returnLast,

I wanted to redirect my new Users after they created their Accounts with User Confirmation activated to another page.

but it only threw them on the main page. 

I looked up why -> in the AbstractController  by creating the target with 

`
if ($this->contentObject->cObjGetSingle(
					$this->config[$action . '.'][$category],
					$this->config[$action . '.'][$category . '.'])
			) {
            $target = $this->contentObject->cObjGetSingle(
                $this->config[$action . '.'][$category],
                $this->config[$action . '.'][$category . '.']
          );
        }

`

it gets filled with an <a href="linktopage you gave an PID for">linktitle</a>

Later when we would create out of this an uri 

`
 if ($target) {
            $this->uriBuilder->setTargetPageUid((int)$target);
            $this->uriBuilder->setLinkAccessRestrictedPages(true);
            $link = $this->uriBuilder->build();
            $this->redirectToUri(StringUtility::removeDoubleSlashesFromUri($link));
        }

`

It wont be able to setTargetPageUid because target cant be cast to an int , so the link object would be empty. And we will get 
redirected to our main Page. 

In my case i configured the code to : 
`
 if ($this->config[$action . '.'][$category . '.']['typolink.']['parameter']
        ) { 
          $target =$this->config[$action . '.'][$category . '.']['typolink.']['parameter'];
                }
`

this works for me in this case but maybe their would be another better way.

**Comment by giesenmartin** [comment]
I have made exactly the same observation as Greg.
Strangely enough, it had already worked in the same installation, and suddenly no longer. 
That's bizzare as I understand it, the redirectByAction method can never work. 
How is this thought of and what could get in the way. 

I use Typo3 10.4.12
My Setup

plugin.tx_femanager.settings.new.forceValues.beforeAnyConfirmation.usergroup = TEXT
plugin.tx_femanager.settings.new.forceValues.beforeAnyConfirmation.usergroup.value = 1

plugin.tx_femanager {
  settings {
    new {
       requestRedirect = TEXT
       requestRedirect {
         typolink {
          parameter = 21
          returnLast = url
          linkAccessRestrictedPages = 1
              }
       }
      userConfirmationRedirect = TEXT
      userConfirmationRedirect {
        typolink {
          parameter = 9
          returnLast = url
          linkAccessRestrictedPages = 1
          }
      }
    }
  }
}

**Comment by cdaecke** [comment]
I can confirm this issue with TYPO3 v10.4.13 and ext:femanager v6.1.2.

The fix of @greggandorf works for me as well.

**Comment by cdaecke** [comment]
Related: #252, #270

**Comment by martinschoene** [comment]
Hello,
i had the same problem with TYPO3 10.4.14 and femanger  6.1.2
and the fix from @greggandorf  works for me as well.

greetings martin

**Comment by spoonerWeb** [comment]
Please check if this works with latest merged commit #258. @rowild @cdaecke @greggandorf @giesenmartin @martinschoene

**Comment by sbusemann** [comment]
It seems to be resolved. If not. please provide feedback.
